airbyte-integrations/connectors/source-jira/source_jira/source.py

```diff
@@ -90,18 +93,29 @@ def get_authenticator(config: Mapping[str, Any]):
 
     def check_connection(self, logger: AirbyteLogger, config: Mapping[str, Any]) -> Tuple[bool, Optional[Any]]:
         try:
+            original_config = config.copy()
             config = self._validate_and_transform(config)
             authenticator = self.get_authenticator(config)
             kwargs = {"authenticator": authenticator, "domain": config["domain"], "projects": config["projects"]}
-            labels_stream = Labels(**kwargs)
-            next(read_full_refresh(labels_stream), None)
+
             # check projects
             projects_stream = Projects(**kwargs)
             projects = {project["key"] for project in read_full_refresh(projects_stream)}
             unknown_projects = set(config["projects"]) - projects
             if unknown_projects:
                 return False, "unknown project(s): " + ", ".join(unknown_projects)
-            return True, None
+
+            # Get streams to check access to any of them
+            streams = self.streams(original_config)
+            for stream in streams:
+                try:
+                    next(read_full_refresh(stream), None)
+                except:
+                    logger.warning("No access to stream: " + stream.name)
+                else:
+                    logger.info(f"API Token have access to stream: {stream.name}, so check is successful.")
+                    return True, None
+            return False, "This API Token does not have permission to read any of the resources."
```

Comment:
do we really need to try read all streams on check? this is certified connector and check command should run fast. we can check only known permissions/scopes, and availability strategy will work while reading and  skip stream if it is not available.

Reply:
@darynaishchenko The updated check method now passes if at least one stream is readable.